doc/tutorials/using_celadon_as_uos.rst

```diff
@@ -0,0 +1,97 @@
+  in order to enable USB device function through the internal USB 3.0 port header.
+* A USB dongle (e.g. `Dawson Canyon USB 3.0 female
+  to 10-pin header cable <https://www.gorite.com/dawson-canyon-usb-3-0-female-to-10-pin-header-cable>`_)
+  is optional if you plan to use the ``adb`` and ``fastboot`` tools in the Celadon User OS for debugging.
+  Refer to the `Technical Product Specification
+  <https://www.intel.com/content/dam/support/us/en/documents/mini-pcs/nuc-kits/NUC7i5DN_TechProdSpec.pdf>`_
+  to identify the USB 3.0 port header on the main board.
+* Two HDMI monitors
+
+Build Celadon from source
+*************************
+
+#. Follow the instructions in the `Build Celadon from source
+   <https://01.org/projectceladon/documentation/getting_started/build-source>`_ guide
+   to set up the Celadon project source code.
+
+#. Select Celadon build target::
+
+      $ cd <Celadon project directory>
+      $ source build/envsetup.sh
+      $ lunch cel_apl-userdebug
+
+   .. note:: You can run ``lunch`` with no arguments to manually choose your Celadon build variants.
+
+#. Download these additional patches and apply each one individually with the command::
+
+       $ git apply <patch-filename>
+
+   .. table:: ACRN patch list
+      :widths: auto
+      :name: ACRN patch list
+
+      +--------------------------------------------------------------------+-------------------------------------------+
+      | Patch link                                                         | Description                               |
+      +====================================================================+===========================================+
+      | https://github.com/projectceladon/device-androidia/pull/458        | kernel config: Add the support of ACRN    |
+      +--------------------------------------------------------------------+-------------------------------------------+
+      | https://github.com/projectceladon/device-androidia-mixins/pull/293 | graphic/mesa: Add the support of ACRN     |
+      +--------------------------------------------------------------------+-------------------------------------------+
+      | https://github.com/projectceladon/device-androidia/pull/441        | cel_apl: use ttyS0 instead of ttyUSB0     |
+      +--------------------------------------------------------------------+-------------------------------------------+
+      | https://github.com/projectceladon/device-androidia/pull/439        | Disable trusty and pstore                 |
+      +--------------------------------------------------------------------+-------------------------------------------+
+
+   .. note:: If the ``git apply`` command shows an error, you may need to modify
+      the source code manually instead.
+
+#. Build Celadon image::
+
+   $ device/intel/mixins/mixin-update
+   $ make SPARSE_IMG=true gptimage -j $(nproc)
+
+.. note:: Replace the ``$(nproc)`` argument with the number of processor threads on your workstation
```

Comment:
You don't need to replace `$(nproc)`, it will be done automatically when you run the `make` command (try to run `echo $(nproc)` on your Linux system if you want to see it in action).

Reply:
I really didn't know the meaning of ``$(nproc)``, now I get it. thank you so much @gvancuts :smile: